src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentCacheGenerationService.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Publisher.Services.Interfaces;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+
+namespace GovUk.Education.ExploreEducationStatistics.Publisher.Services
+{
+    public class ContentCacheGenerationService : IContentCacheGenerationService
+    {
+        private readonly IContentService _contentService;
+        private readonly IReleaseService _releaseService;
+        private readonly IPublicationService _publicationService;
+        private readonly IDownloadService _downloadService;
+        private readonly IMethodologyService _methodologyService;
+        private readonly CloudBlobContainer _cloudBlobContainer;
+
+        private const string ContainerName = "cache";
+
+        public ContentCacheGenerationService(IDownloadService downloadService, IMethodologyService methodologyService,
+            IContentService contentService, IReleaseService releaseService, IPublicationService publicationService,
+            IConfiguration config)
+        {
+            _contentService = contentService;
+            _releaseService = releaseService;
+            _publicationService = publicationService;
+            _downloadService = downloadService;
+            _methodologyService = methodologyService;
+
+            _cloudBlobContainer = GetCloudBlobContainer(config.GetConnectionString("PublicStorage"));
+        }
+
+        public async Task<bool> CleanAndRebuildFullCache()
```

Comment:
I'm not sure about the return type here. Currently it may as well  be void because the methods that are called in turn only ever return true or throw an exception!